### PR TITLE
Partner Portal: Add Billing Dashboard

### PR DESCRIPTION
**Pull request by atanas-dev**
Blocked by
- ~#51906~
- ~D60122-code~
Context: p1HpG7-bsW-p2

#### Changes proposed in this Pull Request

* Add a Billing dashboard to the Partner Portal.

#### Testing instructions

* If you do not have a partner key, please contact Infinity for one or you will be unable to view test this PR.
* Check out this PR locally, then run `yarn && yarn start-jetpack-cloud`.
* Make sure data matches the expected results for the selected partner key.

Screenshot 2021-04-06 at 18 46 39

### CODE REVIEW

File: client/jetpack-cloud/sections/partner-portal/billing-details/index.tsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+						{ ! billing.isSuccess && <TextPlaceholder /> }
+					</span>
+					<strong className="billing-details__cost-amount">
+						{ billing.isSuccess && formatCurrency( billing.data.costs.total, 'USD' ) }
+
+						{ ! billing.isSuccess && <TextPlaceholder /> }
+					</strong>
+
+					<span className="billing-details__total-label billing-details__line-item-meta">
+						{ translate( 'Assigned licenses:' ) }
```
Comment (a8ci18n) [inline client/jetpack-cloud/sections/partner-portal/billing-details/index.tsx:131]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **9** times:
`translate( 'No assigned licenses.' )` *ES Score: 10*

File: client/jetpack-cloud/sections/partner-portal/billing-details/index.tsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+					</span>
+					<span className="billing-details__line-item-meta">
+						{ billing.isSuccess && formatCurrency( billing.data.costs.assigned, 'USD' ) }
+
+						{ ! billing.isSuccess && <TextPlaceholder /> }
+					</span>
+
+					<span className="billing-details__total-label billing-details__line-item-meta">
+						{ translate( 'Unassigned licenses:' ) }
```
Comment (a8ci18n) [inline client/jetpack-cloud/sections/partner-portal/billing-details/index.tsx:140]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **9** times:
`translate( 'No unassigned licenses.' )` *ES Score: 12*

File: client/jetpack-cloud/sections/partner-portal/billing-summary/index.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+				<span className="billing-summary__label">{ translate( 'Total licenses' ) }</span>
+				<strong className="billing-summary__value">
+					{ counts.isSuccess && numberFormat( counts.data.revoked + counts.data.not_revoked, 0 ) }
+
+					{ ! counts.isSuccess && <TextPlaceholder /> }
+				</strong>
+			</div>
+
+			<div className="billing-summary__stat billing-summary__assigned-licenses">
+				<span className="billing-summary__label">{ translate( 'Assigned licenses' ) }</span>
```
Comment (a8ci18n) [inline client/jetpack-cloud/sections/partner-portal/billing-summary/index.tsx:82]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **9** times:
`translate( 'No assigned licenses.' )` *ES Score: 10*

File: client/jetpack-cloud/sections/partner-portal/billing-summary/index.tsx [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+				<strong className="billing-summary__value">
+					{ counts.isSuccess && numberFormat( counts.data.attached, 0 ) }
+
+					{ ! counts.isSuccess && <TextPlaceholder /> }
+				</strong>
+			</div>
+
+			<div className="billing-summary__stat billing-summary__unassigned-licenses">
+				<span className="billing-summary__label">{ translate( 'Unassigned licenses' ) }</span>
```
Comment (a8ci18n) [inline client/jetpack-cloud/sections/partner-portal/billing-summary/index.tsx:91]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **9** times:
`translate( 'No unassigned licenses.' )` *ES Score: 12*

File: client/state/partner-portal/licenses/hooks/use-billing-dashboard-query.ts [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+	const activeKeyId = useSelector( getActivePartnerKeyId );
+
+	return useQuery< APIBilling, TError, Billing >(
+		[ 'partner-portal', 'billing-dashboard', activeKeyId ],
+		queryBillingDashboard,
+		{
+			select: selectBillingDashboard,
+			onError: () => {
+				dispatch(
+					errorNotice( translate( 'We were unable to retrieve your billing details.' ), {
```
Comment (a8ci18n) [inline client/state/partner-portal/licenses/hooks/use-billing-dashboard-query.ts:109]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'We were unable to retrieve your partner account details.' )` *ES Score: 7*

**Review by IanRamosC — APPROVED** [review]
It tests well! Feel free to merge it when the API is online.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/5728730

Thank you @atanas-dev for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/5728730).

**Pull request merged**